src/tcp_wrap.cc

```diff
@@ -325,14 +347,22 @@ void TCPWrap::Connect(const FunctionCallbackInfo<Value>& args,
   T addr;
   int err = uv_ip_addr(*ip_address, &addr);
 
+  wrap->noDelay = args[3]->IsTrue();
+  wrap->keepAlive = args[4]->IsTrue();
```

Comment:
Can we attach the fields to the `ConnectWrap` instead?

In this way it will not increase the memory footprint for each connection.

Reply:
Sure, I can move them there so they are then discarded after connection.

@addaleax Any objection on using this approach? Or do you think calling JS methods will not impact performance too much?

Reply:
@ShogunPanda Well yeah, @mcollina’s suggestion is definitely something that would improve this PR :slightly_smiling_face: 

> Or do you think calling JS methods will not impact performance too much?

I would still think that that’s absolutely preferable, and not expect significant performance overhead, because:

1. These calls are performed once-per-connection, which is much less than e.g. the number of data events for typical sockets, so it will have comparably little impact, and
2. Doing JS-only changes would still allow us to be just as fast in the common case (by calling `uv_tcp_nodelay(handle, 0);` in the connect handler, and only doing another call into C++ when the user explicitly requested that the value is *not* set)

Generally we try to stick to implementing as much logic in JS as possible, because it is the more accessible language of the two that Node.js is made out of :slightly_smiling_face: